codec-http2/src/test/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionEncoderTest.java

```diff
@@ -200,20 +201,6 @@ public Object answer(InvocationOnMock invocationOnMock) throws Throwable {
     }
 
     @Test
-    public void dataWriteAfterGoAwayShouldFail() throws Exception {
```

Comment:
@Scottmitch 

> 1. We can send HEADERS and DATA after a GO_AWAY is sent (mocking the appropriate methods to make it look like we sent a goaway).

That is, setting the `lastStreamId`, closing every stream > `lastStreamId` ... sounds a lot like the actual implementation. crazy idea. why don't we test our actual implementation instead of mocks for change? :P

> 2.We can not send a HEADERS and DATA after a GO_AWAY is received AND the lastStreamId > what was in the GO_AWAY frame we received.

That is, setting the `lastStreamId`, closing every stream > `lastStreamId` ... sounds a lot like the actual implementation ....

I believe I said it before, I think all the unit tests in the HTTP/2 codec are mocked way too much. Some of it is a matter of taste, but I think things like this are a prime example of overusing mocking.